### PR TITLE
feat: task: add resolver_usernames to POST /task

### PR DESCRIPTION
**Pull request by rbeuque74**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

* **What is the new behavior (if this is a feature change)?**
Adding `resolver_usernames` as input for `POST /task` to allow admins and template owners to add more resolvers to a task.
This allow to give control to more users over a task. Usefull in case the task is created by an automated process, and we want to let users control over the task execution.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Other information**:

### CODE REVIEW

File: api/handler/task.go [hunk 1]
```diff
@@ -62,7 +63,22 @@ func CreateTask(c *gin.Context, in *createTaskIn) (*task.Task, error) {
...
+		// as the template might have defined a limited set of users that can resolve the task.
+		// We need to be sure that the requester will not grant himself (or anybody else) as resolver
+		// and bypass the resolver restriction set on the task_template.
+		// Let's limit this parameter to template owner and admins.
+
+		admin := auth.IsAdmin(c) == nil
+		templateOwner := auth.IsTemplateOwner(c, tt) == nil
+
+		if !admin && !templateOwner {
+			return nil, errors.Forbiddenf("resolver_usernames can't be set as regular user, you need to be owner of the template, or admin")
```
Comment (Alkorin) [inline api/handler/task.go:77]:
```suggestion
			return nil, errors.Forbiddenf("resolver_usernames can't be set by a regular user, you need to be owner of the template or admin")
```

File: api/handler/task.go [hunk 2]
```diff
@@ -62,7 +63,22 @@ func CreateTask(c *gin.Context, in *createTaskIn) (*task.Task, error) {
 		return nil, err
 	}
 
-	t, err := taskutils.CreateTask(c, dbp, tt, in.WatcherUsernames, []string{}, in.Input, nil, in.Comment, in.Delay, in.Tags)
+	if len(in.ResolverUsernames) > 0 {
+		// if user is neither admin or template owner, prevent to set the resolver_usernames
+		// as the template might have defined a limited set of users that can resolve the task.
+		// We need to be sure that the requester will not grant himself (or anybody else) as resolver
+		// and bypass the resolver restriction set on the task_template.
+		// Let's limit this parameter to template owner and admins.
```
Comment (Alkorin) [inline api/handler/task.go:71]:
```suggestion
		// if user is neither admin nor template owner, prevent setting the resolver_usernames
		// as the template might have defined a limited set of users that can resolve the task.
		// We need to be sure that the requester will not grant himself (or anybody else) as resolver
		// and bypass the resolver restriction set on the task_template.
		// Let's limit this parameter to template owners and admins.
```

File: api/handler/task.go (hunk 1 above)
Comment (rbeuque74) [reply api/handler/task.go]: Thanks, applied

File: api/handler/task.go (hunk 2 above)
Comment (rbeuque74) [reply api/handler/task.go]: Thanks, applied

**Pull request merged**